tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/internal/types/Tensors.java

```diff
@@ -0,0 +1,158 @@
+  }
+
+  /**
+   * Equivalent to {@link #toString(Tensor, Integer) toString(tensor, null)}.
+   *
+   * @param tensor a tensor
+   * @return the String representation of the tensor
+   */
+  public static String toString(Tensor tensor) {
+    return toString(tensor, null);
+  }
+
+  /**
+   * @param tensor   a tensor
+   * @param maxWidth the maximum width of the output in characters ({@code null} if unlimited). This
+   *                 limit may surpassed if the first or last element are too long.
+   * @return the String representation of the tensor
+   */
+  public static String toString(Tensor tensor, Integer maxWidth) {
+    if (!(tensor instanceof NdArray)) {
+      throw new AssertionError("Expected tensor to extend NdArray");
+    }
+    NdArray<?> ndArray = (NdArray<?>) tensor;
+    Iterator<? extends NdArray<?>> iterator = ndArray.scalars().iterator();
+    Shape shape = tensor.shape();
+    if (shape.numDimensions() == 0) {
+      if (!iterator.hasNext()) {
+        return "";
+      }
+      return String.valueOf(iterator.next().getObject());
+    }
+    return toString(iterator, shape, 0, maxWidth);
+  }
+
+  /**
+   * @param iterator  an iterator over the scalars
+   * @param shape     the shape of the tensor
+   * @param maxWidth  the maximum width of the output in characters ({@code null} if unlimited).
+   *                  This limit may surpassed if the first or last element are too long.
+   * @param dimension the current dimension being processed
+   * @return the String representation of the tensor data at {@code dimension}
+   */
+  private static String toString(Iterator<? extends NdArray<?>> iterator, Shape shape,
+      int dimension, Integer maxWidth) {
+    if (dimension < shape.numDimensions() - 1) {
+      StringJoiner joiner = new StringJoiner(",\n", indent(dimension) + "[\n",
+          "\n" + indent(dimension) + "]");
+      for (long i = 0, size = shape.size(dimension); i < size; ++i) {
+        String element = toString(iterator, shape, dimension + 1, maxWidth);
+        joiner.add(element);
+      }
+      return joiner.toString();
+    }
+    if (maxWidth == null) {
+      StringJoiner joiner = new StringJoiner(", ", indent(dimension) + "[", "]");
+      for (long i = 0, size = shape.size(dimension); i < size; ++i) {
+        String element = iterator.next().getObject().toString();
+        joiner.add(element);
+      }
+      return joiner.toString();
+    }
+    List<Integer> lengths = new ArrayList<>();
+    StringJoiner joiner = new StringJoiner(", ", indent(dimension) + "[", "]");
+    int lengthBefore = "]".length();
+    for (long i = 0, size = shape.size(dimension); i < size; ++i) {
+      String element = iterator.next().getObject().toString();
+      joiner.add(element);
+      int addedLength = joiner.length() - lengthBefore;
+      lengths.add(addedLength);
+      lengthBefore += addedLength;
+    }
+    return truncateWidth(joiner.toString(), maxWidth, lengths);
+  }
+
+  /**
```

Comment:
Can you add a basic description of what this does?

Reply:
I'm confused. The `truncateWidth()` methods is already documented. What did you mean?

Reply:
I've added a description. Please review.